### PR TITLE
Implement test case status preview; #459

### LINKED ISSUE #459: Jump to failed testcase
Filing on behalf of @FatalEagle, something among the lines of:

image

Would allow to jump directly to failed cases in long problems. How this would work with batches interleaved with regular testcases has yet to be decided, but one idea is to omit long contiguous sections of AC regular testcases (`1 2 3 4 ... 300 301(WA) 302 ... 500`). Batches would be treated the same way, though they would always be displayed (interrupting a `...` section), and have a superscript identifying them as batches (e.g., `15/15` for AC, and `X/15`) for WA.

### PR DESCRIPTION
**Pull request by Ninjaclasher**
Closes #459.

**Screenshots from a test site:**
Screenshot from 2020-03-26 14-20-09
Screenshot from 2020-03-26 14-20-15

**Screenshots from a production site (MCPT). Note that the icon sizes are slightly different due to an upgraded Font Awesome set:**
Screenshot from 2020-03-26 14-22-33
Screenshot from 2020-03-26 14-22-57
Screenshot from 2020-03-26 14-23-48

### CODE REVIEW

**Comment by Ninjaclasher** [comment]
Added a commit to combine any consecutive status code (not just AC's) to fix this:
Screenshot from 2020-03-26 15-31-39

**Comment by Xyene** [comment]
Nice mockup from @ehhthing:

image

**Comment by Xyene** [comment]
The highlight isn't quite contiguous.

image

**Comment by Ninjaclasher** [comment]
Larry's mockup has been implemented.

File: templates/submission/status-testcases.html [hunk 1]
```diff
@@ -33,7 +53,7 @@ <h3>{{ _('Execution Results') }}</h3>
                 <br>
                 <div class="batch-cases">
             {% endif %}
-        <table>{% for case in batch.cases %}
+        <table id="submissions-status-table">{% for case in batch.cases %}
```
Comment (Xyene) [inline templates/submission/status-testcases.html:56]: `id` is meant to be globally unique, whereas a submission may have multiple batches.

**Pull request merged**